src/olympia/addons/models.py

```diff
@@ -1305,6 +1305,17 @@ def all_categories(self):
         return filter(
             None, [cat.to_static_category() for cat in self.categories.all()])
 
+    @cached_property
+    def current_previews(self):
+        """Previews for the current version, or all of them if not a
+        static theme."""
+        if self.type == amo.ADDON_STATICTHEME:
+            if self.current_version:
+                return self.current_version.previews.all()
+            return []
+        else:
+            return self.all_previews
```

Comment:
I've been thinking: in https://github.com/mozilla/addons-server/issues/7756 I didn't make any assumption about Addon.type and just deleted any Previews for the Addon and any VersionPreviews for the Versions - maybe we should just `return list(self.current_version.previews.all()) + self.all_previews` for all add-ons?   

I'm guessing Serializer problems with ES though...

Reply:
That would require some extra optimizations to avoid extra queries though, even without caring about the ES serializer.